rust/definitions/src/history.rs

```diff
@@ -256,38 +229,40 @@ impl TypesExport {
             signed_by: signed_by.to_owned(),
         }
     }
-
-    /// Print json with [`TypesExport`] information for user interface
-    #[cfg(feature = "signer")]
-    pub fn show(&self) -> String {
-        let types_id_pic = hex::encode(pic_types(&self.types_hash));
-        format!(
-            "\"types_hash\":\"{}\",\"types_id_pic\":\"{}\",\"signed_by\":{}",
-            hex::encode(&self.types_hash),
-            types_id_pic,
-            export_complex_single(&self.signed_by, |a| a.show_card())
-        )
-    }
 }
 
-/// Event content for address generation or removal
-///
-/// Contains public information associated with address:
-/// - seed name  
-/// - [`Encryption`]  
-/// - public key for address  
-/// - path with soft (`/`) and hard (`//`) derivatinos only, **without** password  
-/// - genesis hash of the network within which the address is  
+/// Event content for address generation or removal.
 #[derive(Debug, Decode, Encode, PartialEq, Clone)]
 pub struct IdentityHistory {
-    seed_name: String,
-    encryption: Encryption,
-    public_key: Vec<u8>,
-    path: String,
-    network_genesis_hash: Vec<u8>,
+    /// The name of the seed.
+    pub seed_name: String,
+    /// [`Encryption`] scheme of the seed.
+    pub encryption: Encryption,
+    /// Public key.
+    pub public_key: Vec<u8>,
+    /// - path with soft (`/`) and hard (`//`) derivatinos only, **without** password  
+    pub path: String,
+    /// - genesis hash of the network within which the address is  
+    pub network_genesis_hash: Vec<u8>,
```

Comment:
And here H256 too. And further on.
Also, not sure if we should do it, but meta_hash and types_hash are 32 bytes as defined, so there are no other length options for them either.